### PR TITLE
Rename classes to avoid name clashes

### PR DESCRIPTION
**Pull request by xirc**
This PR solves the issue https://github.com/ghostdogpr/caliban/issues/543.

https://github.com/ghostdogpr/caliban/issues/543#issuecomment-670833351
> How about renaming those internal classes to __???Value directly?

As suggested by @ghostdogpr, I've changed the internal class names.

- NullValue to __NullValue
- NumberValue to __NumberValue
- EnumValue to __EnumValue
- StringValue to __StringValue
- BooleanValue to __BooleanValue
- ListValue to __ListValue
- ObjectValue to __ObjectValue

### CODE REVIEW

**Comment by ghostdogpr** [comment]
Nice! There's one more thing to do: modifying the code generation to use those new names. The relevant code is there: https://github.com/ghostdogpr/caliban/blob/master/tools/src/main/scala/caliban/tools/ClientWriter.scala 

Let me know if you need more explanations.

**Comment by xirc** [comment]
@ghostdogpr Thanks for the review! I will try it tomorrow morning (JST+9:00).
Because the CI is success, I guess that the code generation is not tested yet, right?
I think it's better to test the code generation.
So, I will also try to test it if there is no technical problem.

**Comment by ghostdogpr** [comment]
It is tested (here: https://github.com/ghostdogpr/caliban/blob/master/tools/src/test/scala/caliban/tools/ClientWriterSpec.scala) but the output is a string (generated code), and we just verify that this is the expected code, we don't actually verify that it compiles. That's why the tests passed. If you have a better idea for testing this, feel free to add it 👍

**Comment by xirc** [comment]
Thanks! I got it.
In this PR, I will focus on changing the code generation and its existing test because I think writing a test verifying that generated code compiles is out of topic in this PR and makes the PR huge.

File: client/src/main/scala/caliban/client/Value.scala [hunk 1]
```diff
@@ -8,47 +8,47 @@ import io.circe.{ Decoder, Encoder, Json }
 sealed trait Value
```
Comment (xirc) [inline client/src/main/scala/caliban/client/Value.scala:8]: Is it better to rename `Value` to `__Value`?
I think the type may have the potential to collide with a user type name.

Comment (ghostdogpr) [reply client/src/main/scala/caliban/client/Value.scala:8]: Yeah, I agree.

**Review by ghostdogpr — APPROVED** [review]
Perfect! Thanks for the contribution!

**Comment by jeffrey-aguilera** [comment]
Ugg. This broke my build. Should call out breaking changes like this in the release notes.

**Pull request merged**